README.md
<div align='center'>

  <h1>AppSpace</h1>

  <h4>
    私有化部署服务,集成应用分发 Bug管理 提供iOS、Android、Flutter、React-Native SDK，支持自动化部署jenkins fastlane等丰富组件 
  </h4>

  <div>
    <a href="https://appspa.github.io/">私有化部署</a> •
    <a href="https://appspa.github.io/docs/developer-guide">快速接入</a> •
    <a href="https://github.com/appspa/app-space-docker">docker部署</a> •
  </div>
</div>

## 在线演示
> **注意**: 演示服务使用免费资源部署，资源额度耗尽会不可访问次月初重置；数据每日都会重新初始化，不对用户上传的应用承担任何法律风险，后果自负！(翻墙)

- 演示地址：https://app-space.up.railway.app
- 登录账户: `admin`
- 登录密码：`app@space`

## 功能列表

![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_6.png)
### 应用设置
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_4.png)
### 发布应用
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_5.png)
### 团队管理
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_3.png)
### 应用下载
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_2.png)
### 钉钉机器人
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_1.png)

<<<<<<< HEAD
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_6.png)
### 应用设置
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_4.png)
### 发布应用
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_5.png)
### 团队管理
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_3.png)
### 应用下载
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_2.png)
### 钉钉机器人
![AppSpace home](https://github.com/appspa/appspa.github.io/blob/main/static/img/img_1.png)

=======
>>>>>>> patch
# 源代碼部署


### 运行前准备

* 安装 MongoDB (3.6)
* 安装 Nodejs


1.clone 下载代码 `git clone https://github.com/appspa/app-space.git`

2.运行server端

```bash
#进入项目根目录
cd server
npm install  #安装依赖
npm start

=============>>out
...
...
App is listening on 8081.
数据库连接成功
=============>>end
```

3.编译运行前端页面

```Bash
cd client
npm install
npm run build #正式环境可以用该命令编译静态文件交给nginx
npm run start  #本地运行可以使用该命令

============>>out
...
...
 DONE  Compiled successfully in 18546ms                                                
 I  Your application is running here: http://0.0.0.0:8081
============>>end
```

4.使用浏览器打开进入http://localhost:8081即可进入应用




### 项目配置说明

**前端配置**

参见 client/config.js
```javascript
const config = {
  domain: process.env.APP_SPA_DOMAIN || "http://192.168.1.10:8081",
  // basePath: process.env.APP_SPA_PATH || "/x/app/",
};

export default config;

```

**后端配置**

参见 server/config.js

```javascript
//需要修改配置可以修改config.js文件,也可以在部署的时候导出环境变量
//比如 export APP_SPA_DOMAIN=https://127.0.0.1:8085

const common = {
    //baseUrl应用请求的url地址,比如https://appspace
    baseUrl: process.env.APP_SPA_DOMAIN || "https://127.0.0.1:8081", 
    port: process.env.APP_SPA_PORT || "8085", //server运行的端口
    apiPrefix: 'api',
    secret: process.env.APP_SPA_SECRET || "secretsecret", //secret
    //数据库用户 (没有开启mongodb用户认证的可以不填写)
    dbUser: process.env.APP_SPA_DBUSER || undefined,  
    //数据库密码 (没有开启mongodb用户认证的可以不填写)
    dbPass: process.env.APP_SPA_DBPWD || undefined,  
    dbName: process.env.APP_SPA_DB_NAME || "app_space", //数据库名称
    dbHost: process.env.APP_SPA_DB_HOST || "localhost", //数据库地址
    dbPort: process.env.APP_SPA_DB_PORT || "27017", //数据库端口
	
    //邮件相关配置 用于找回密码和邀请团队成员发送邮件
    emailService: process.env.APP_SPA_EMAIL_SERVICE || "qq", 
    emailUser: process.env.APP_SPA_EMAIL_USER || "", 
    emailPass: process.env.APP_SPA_EMAIL_PASS || "",

    //是否允许用户注册,为否则后端注册接口不可用
    allowRegister: process.env.APP_SPA_ALLOW_REGISTER || true, 

    //是否开启ldap 默认是false 如果公司没有ldap服务可以不用理会
    openLdap: process.env.APP_SPA_ALLOW_LDAP || false, 
    ldapServer: process.env.APP_SPA_LDAP_URL || "",  //ldap server url
    ldapUserDn: process.env.APP_SPA_LDAP_USERDN || "", //ldap管理员dn 管理员用户名
    ldapBindCredentials: process.env.APP_SPA_LDAP_CREDENTIALS || "", //ldap管理员密码
    ldapBase: process.env.APP_SPA_LDAP_BASE || "" //ldap base

};
```





### 正式环境部署 nginx配置(注意请使用https部署,否则iOS会出现无法安装的问题)

可以按照项目根目录的 app_space_nginx.conf 文件进行配置

```bash
server{
  listen 80;
  server_name appspace;

  #root目录为项目根目录的client/dist目录下,前端静态页面
  root /home/ubuntu/appspace/client/dist;
  index index.html;

  location / {
      try_files $uri $uri/ @router;
      index index.html;
  }

  location /api/ {  #把以api打头的接口转发给后端server
    proxy_pass http://127.0.0.1:8085; #这里端口修改为后端服务运行的端口
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
  client_max_body_size 208M; #最大上传的ipa/apk文件大小
}
```
[//]: # ()
[//]: # (## 在线演示)

[//]: # ()
[//]: # (## 发布协议)
## 联系作者
QQ:1206149686

<img src="https://github.com/appspa/appspa.github.io/blob/main/static/img/qq.png" height=120/>

[MIT][mit-link]


[app-space-ios-sdk]: https://github.com/appspa/app-space-ios-sdk
[app-space-android-sdk]: https://github.com/appspa/app-space-android-sdk
[fastlane-plugin-app-space]: https://github.com/appspa/fastlane-plugin-app-space
[mit-link]: https://github.com/appspa/app-space/blob/develop/CHANGELOG.md


CHANGELOG.md
1.railway s

server/config.js
import os from "os";

let fs = require('fs');
let path = require('path');


const config = {
  // baseUrl: process.env.APP_SPA_DOMAIN || "http://127.0.0.1:80", //baseUrl应用请求的url地址,比如https://appspace
  // port: process.env.APP_SPA_PORT || '80', //server运行的端口
  baseUrl: process.env.APP_SPA_DOMAIN || "http://127.0.0.1:8081", //baseUrl应用请求的url地址,比如https://appspace
  host: process.env.APP_SPA_HOST || "0.0.0.0", //host railway
  port: process.env.APP_SPA_PORT || '8081', //server运行的端口

  iosPlistSource: 'api', //[api,yun] api通过接口获取 yun通过云盘接口获取（前提有云盘存储） 解决ios本地无https接口 用云盘方案
  //解决局域网iOS https证书 临时体验用
  iosInstallUrl: undefined, // "https://127.0.0.1:443"
  // iosInstallPort: "443",//undefined
  apiPrefix: 'api',
  secret: process.env.APP_SPA_SECRET || 'secret', //secret
  dbUser: process.env.APP_SPA_DBUSER || 'mongo', //数据库用户 (没有开启mongodb用户认证的可以不填写)
  dbPass: process.env.APP_SPA_DBPWD || 'bWHy6ZBakeOx0oMV3fG', //数据库密码 (没有开启mongodb用户认证的可以不填写)
  dbName: process.env.APP_SPA_DB_NAME || 'app_space',//'app_space', //数据库名称
  dbHost: process.env.APP_SPA_DB_HOST ||'containers-us-west-181.railway.app',// 'mongo',// '127.0.0.1', //数据库地址

  dbPort: process.env.APP_SPA_DB_PORT || '6106', //数据库端口

  emailService: process.env.APP_SPA_EMAIL_SERVICE || 'qq', //邮件相关配置 用于找回密码和邀请团队成员发送邮件
  emailUser: process.env.APP_SPA_EMAIL_USER || '',
  emailPass: process.env.APP_SPA_EMAIL_PASS || '',
  emailPort: process.env.APP_SPA_EMAIL_PORT || '587',

  allowRegister: boolConfig(process.env.APP_SPA_ALLOW_REGISTER || 'false'), //是否允许用户注册,为否则后端注册接口不可用
  adminAccount: process.env.APP_SPA_ADMIN_ACCOUNT || 'admin',
  adminPassword: process.env.APP_SPA_ADMIN_PASSWORD || 'app@space',

  openLdap: boolConfig(process.env.APP_SPA_ALLOW_LDAP || 'false'), //是否开启ldap 默认是false 如果公司没有ldap服务可以不用理会
  ldapServer: process.env.APP_SPA_LDAP_URL || '', //ldap server url
  ldapUserDn: process.env.APP_SPA_LDAP_USERDN || '', //ldap管理员dn 也就是管理员用户名
  ldapBindCredentials: process.env.APP_SPA_LDAP_CREDENTIALS || '', //ldap管理员密码
  ldapBase: process.env.APP_SPA_LDAP_BASE || '', //ldap base

  storageType: process.env.STORAGE_TYPE || 'local', //"s3",
  storageDir: process.env.STORAGE_DIR || '/data/storage',//path.join(__dirname, '../docker/storage'),

  // Config for local storage when storageType value is "local".
  local: {
    // downloadUrl
    // downloadUrl: process.env.APP_SPA_DOMAIN || "http://10.151.100.79:80",
    downloadUrl: process.env.APP_SPA_DOMAIN || "http://192.168.1.10:8085",
  },
  // Config for qiniu (http://www.qiniu.com/) cloud storage when storageType value is "qiniu".
  qiniu: {
    accessKey: process.env.QIN_IU_ACCESS_KEY,
    secretKey: process.env.QIN_IU_SECRET_KEY,
    bucketName: process.env.QIN_IU_BUCKET_NAME,
    downloadUrl: process.env.QIN_IU_DOWNLOAD_URL // Binary files download host address.
  },
  // Config for upyun (https://www.upyun.com/) storage when storageType value is "upyun"
  upyun: {
    storageDir: process.env.UP_YUN_STORAGE_DIR,
    serviceName: process.env.UP_YUN_SERVICE_NAME,
    operatorName: process.env.UP_YUN_OPERATOR_NAME,
    operatorPass: process.env.UP_YUN_OPERATOR_PASS,
    downloadUrl: process.env.UP_YUN_DOWNLOAD_URL,
  },

  // Config for Amazon s3 (https://aws.amazon.com/cn/s3/) storage when storageType value is "s3".
  s3: {
    accessKeyId: process.env.AWS_ACCESS_KEY_ID || '',
    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY || '',
    sessionToken: process.env.AWS_SESSION_TOKEN || '', //(optional)
    bucketName: process.env.AWS_BUCKET_NAME || 't',
    region: process.env.AWS_REGION || '',
    downloadUrl: process.env.AWS_DOWNLOAD_URL||'', // binary files download host address.
  },
  // Config for tencentyun COS (https://cloud.tencent.com/product/cos) when storageType value is "oss".
  tencentcloud: {
    accessKeyId: process.env.TENCENT_CLOUD_ACCESS_KEY_ID,
    secretAccessKey: process.env.TENCENT_CLOUD_SECRET_ACCESS_KEY,
    bucketName:process.env.TENCENT_CLOUD_BUCKET_NAME,
    region: process.env.TENCENT_CLOUD_REGION,
    downloadUrl: process.env.TENCENT_CLOUD_DOWNLOAD_URL, // binary files download host address.
  },
  // Config for Aliyun OSS (https://www.aliyun.com/product/oss) when storageType value is "oss".
  oss: {
    accessKeyId:  process.env.OSS_ACCESS_KEY_ID,
    secretAccessKey:  process.env.OSS_SECRET_ACCESS_KEY,
    endpoint:  process.env.OSS_ENDPOINT,
    bucketName:  process.env.OSS_BUCKET_NAME,
    prefix:  process.env.OSS_PREFIX, // Key prefix in object key
    downloadUrl:  process.env.OSS_DOWNLOAD_URL, // binary files download host address.
  },

  jwt: {
    // Recommended: 63 random alpha-numeric characters
    // Generate using: https://www.grc.com/passwords.htm
    tokenSecret: process.env.TOKEN_SECRET || 'INSERT_RANDOM_TOKEN_KEY'
  },

  // Config for smtp email，register module need validate user email project source https://github.com/nodemailer/nodemailer
  smtpConfig: {
    host: "smtp.aliyun.com",
    port: 465,
    secure: true,
    auth: {
      user: "",
      pass: ""
    }
  },
  // Config for redis (register module, tryLoginTimes module)
  redis: {
    default: {
      host: "127.0.0.1",
      port: 6379,
      retry_strategy: function (options) {
        if (options.error.code === 'ECONNREFUSED') {
          // End reconnecting on a specific error and flush all commands with a individual error
          return new Error('The server refused the connection');
        }
        if (options.total_retry_time > 1000 * 60 * 60) {
          // End reconnecting after a specific timeout and flush all commands with a individual error
          return new Error('Retry time exhausted');
        }
        if (options.times_connected > 10) {
          // End reconnecting with built in error
          return undefined;
        }
        // reconnect after
        return Math.max(options.attempt * 100, 3000);
      }
    }
  },
  log4js: {
    appenders: {console: {type: 'console'}},
    categories: {
      "default": {appenders: ['console'], level: 'all'},
      "startup": {appenders: ['console'], level: 'all'},
      "http": {appenders: ['console'], level: 'all'}
    }
  }
};

function boolConfig(str) {
  return str === 'true'
}

export default config;
